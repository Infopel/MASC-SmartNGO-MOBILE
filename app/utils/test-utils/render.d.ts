export type RenderOptions = {
  wrapper?: React.ComponentType<any>
  createNodeMock?: (element: React.ReactElement) => any
}
export type AllTheProvidersProps = { children: React.ReactNode }
