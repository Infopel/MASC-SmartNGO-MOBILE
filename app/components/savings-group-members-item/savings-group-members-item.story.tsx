import * as React from "react"
import { storiesOf } from "@storybook/react-native"
import { StoryScreen, Story, UseCase } from "../../../storybook/views"
import { color } from "../../theme"
import { SavingsGroupMembersItem } from "./savings-group-members-item.component"

storiesOf("SavingsGroupMembersItem", module)
  .addDecorator((fn) => <StoryScreen>{fn()}</StoryScreen>)
  .add("Style Presets", () => (
    <Story>
      <UseCase text="Primary" usage="The primary.">
        <SavingsGroupMembersItem style={{ backgroundColor: color.error }} />
      </UseCase>
    </Story>
  ))
