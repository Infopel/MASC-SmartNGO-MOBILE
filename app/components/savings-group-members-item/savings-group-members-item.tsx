import { BaseListItem } from "components/base-list-item/base-list-item"
import { Button } from "components/button/button"
import { Text } from "components/text/text"
import { HStack, VStack, Button as NBButton } from "native-base"
import * as React from "react"
import { StyleProp, ViewStyle } from "react-native"
import { ISavingsGroupMemberItem } from "storage/queries/savings-group"
import { DateUtils } from "utils/DateUitls"

export interface SavingsGroupMembersItemProps {
  /**
   * An optional style override useful for padding & margin.
   */
  style?: StyleProp<ViewStyle>
  item: ISavingsGroupMemberItem
  handleOpenMonitoringForm: () => void
  handleOpenGiveUpForm: () => void
}

/**
 * Describe your component here
 */
export const SavingsGroupMembersItem = function SavingsGroupMembersItem(props: SavingsGroupMembersItemProps) {
  const { item,handleOpenGiveUpForm,handleOpenMonitoringForm } = props

  return (
    <BaseListItem p={2}>
      <VStack space={2}>
        <HStack space={2}>
          <VStack flex={1} space={2}>
            <Text fontWeight={'bold'}>{item.name}</Text>
            <Text>{item.age} anos</Text>
            <Text>{DateUtils.prettyDate(item.lastMonitoredAt)}</Text>
          </VStack>
          <VStack space={2}>
            <Text fontWeight={'bold'}>Poupança: {item.savings}</Text>
            <Text>Empréstimo: {item.loan} </Text>
          </VStack>
        </HStack>
        <NBButton.Group justifyContent={'flex-end'}>
          <Button onPress={handleOpenMonitoringForm} tx='savings-group-details-members-tab.monitoring'/>
          <Button variant={'outline'} onPress={handleOpenGiveUpForm} tx='savings-group-details-members-tab.quitting'/>
        </NBButton.Group>
      </VStack>
    </BaseListItem>
  )
}
