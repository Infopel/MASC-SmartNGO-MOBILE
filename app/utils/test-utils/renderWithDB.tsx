jest.mock("storage/adapter.ts")
import DatabaseProvider from "@nozbe/watermelondb/DatabaseProvider"
import { render } from "@testing-library/react-native"
import { default as React } from "react"
import { database } from "storage/database"
import { NativeBaseProvider as ThemeProvider } from "theme"
import { AllTheProvidersProps, RenderOptions } from "./render"

function AllTheProvidersWithDB({ children }: AllTheProvidersProps) {
    return (
        <DatabaseProvider database={database}>
            <ThemeProvider>{children}</ThemeProvider>
        </DatabaseProvider>
    )
}

export function renderWithDB<T extends any>(ui: React.ReactElement<T>, options?: RenderOptions) {
    return render(ui, { wrapper: AllTheProvidersWithDB, ...options })
}


export class DiagnosticError extends Error {
    name: string = "Diagnostic error"
  }
  