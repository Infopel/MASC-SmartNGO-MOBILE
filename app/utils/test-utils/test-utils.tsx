import { render } from "@testing-library/react-native"
import React from "react"
import { NativeBaseProvider as ThemeProvider } from "theme"
import { AllTheProvidersProps, RenderOptions } from "./render"

function AllTheProviders({ children }: AllTheProvidersProps) {
  return (
    <ThemeProvider>{children}</ThemeProvider>
  )
}

function customRender<T extends any>(ui: React.ReactElement<T>, options?: RenderOptions) {
  return render(ui, { wrapper: AllTheProviders, ...options })
}

// re-export everything
export * from "@testing-library/react-native"
// override render method
export { customRender as render }

