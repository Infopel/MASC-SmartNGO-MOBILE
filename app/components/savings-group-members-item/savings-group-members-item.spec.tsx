import { fireEvent, render } from "utils/test-utils/test-utils"
import * as React from "react"
import { SavingsGroupMembersItem } from "./savings-group-members-item"



describe("<SavingsGroupMembersItem>", () => {
  it("render", async function () {
    const component = <SavingsGroupMembersItem item={{
      age: 78, id: "asdsad", lastMonitoredAt: 1343433233, loan: 231, name: "Angoche", savings: 15844854
    }}></SavingsGroupMembersItem>

    expect(render(component).toJSON()).toMatchSnapshot()
  })
})
