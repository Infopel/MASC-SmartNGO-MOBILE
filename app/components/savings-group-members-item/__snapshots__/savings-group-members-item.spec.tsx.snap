// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SavingsGroupMembersItem> render 1`] = `
<View
  dataSet={Object {}}
  style={
    Object {
      "backgroundColor": "#fafaf9",
      "borderBottomColor": "#fafafa",
      "borderLeftColor": "#ef8812",
      "borderLeftWidth": 6,
      "borderRadius": 6,
      "borderRightColor": "#fafafa",
      "borderTopColor": "#fafafa",
      "borderWidth": 1,
      "elevation": 2,
      "paddingBottom": 8,
      "paddingLeft": 8,
      "paddingRight": 8,
      "paddingTop": 8,
      "shadowColor": "#000000",
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.2,
      "shadowRadius": 1.41,
    }
  }
>
  <View
    dataSet={Object {}}
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      dataSet={Object {}}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Text
        dataSet={Object {}}
        style={
          Object {
            "backgroundColor": undefined,
            "color": "#171717",
            "fontFamily": undefined,
            "fontSize": 14,
            "fontStyle": "normal",
            "fontWeight": "700",
            "letterSpacing": 0,
            "lineHeight": 21,
            "textDecorationLine": undefined,
          }
        }
      >
        Angoche
      </Text>
      <View
        dataSet={Object {}}
        style={
          Object {
            "height": 8,
          }
        }
      />
      <Text
        dataSet={Object {}}
        style={
          Object {
            "backgroundColor": undefined,
            "color": "#171717",
            "fontFamily": undefined,
            "fontSize": 14,
            "fontStyle": "normal",
            "fontWeight": "400",
            "letterSpacing": 0,
            "lineHeight": 21,
            "textDecorationLine": undefined,
          }
        }
      >
        78
         anos
      </Text>
      <View
        dataSet={Object {}}
        style={
          Object {
            "height": 8,
          }
        }
      />
      <Text
        dataSet={Object {}}
        style={
          Object {
            "backgroundColor": undefined,
            "color": "#171717",
            "fontFamily": undefined,
            "fontSize": 14,
            "fontStyle": "normal",
            "fontWeight": "400",
            "letterSpacing": 0,
            "lineHeight": 21,
            "textDecorationLine": undefined,
          }
        }
      >
        16/01/1970
      </Text>
    </View>
    <View
      dataSet={Object {}}
      style={
        Object {
          "width": 8,
        }
      }
    />
    <View
      dataSet={Object {}}
      style={Object {}}
    >
      <Text
        dataSet={Object {}}
        style={
          Object {
            "backgroundColor": undefined,
            "color": "#171717",
            "fontFamily": undefined,
            "fontSize": 14,
            "fontStyle": "normal",
            "fontWeight": "700",
            "letterSpacing": 0,
            "lineHeight": 21,
            "textDecorationLine": undefined,
          }
        }
      >
        Poupança: 
        15844854
      </Text>
      <View
        dataSet={Object {}}
        style={
          Object {
            "height": 8,
          }
        }
      />
      <Text
        dataSet={Object {}}
        style={
          Object {
            "backgroundColor": undefined,
            "color": "#171717",
            "fontFamily": undefined,
            "fontSize": 14,
            "fontStyle": "normal",
            "fontWeight": "400",
            "letterSpacing": 0,
            "lineHeight": 21,
            "textDecorationLine": undefined,
          }
        }
      >
        Empréstimo: 
        231
         
      </Text>
    </View>
  </View>
</View>
`;
